### PR TITLE
Larry/131 confirm button & input error indication

### PR DESCRIPTION
**Pull request by larryzuo**
Fixes:
#131 

Adds:
- Confirm button with disabled state if the textfields are empty or erroneous
- Error indication for each textfield
- Code refactor & improvements

### CODE REVIEW

**Comment by mathiasquintero** [comment]
The CI for some reason is not handling Swift 4 very well

File: TUM Campus App/LoginViewController.swift [hunk 1]
```diff
@@ -13,29 +13,94 @@ class LoginViewController: UIViewController {
...
+    }
+
+    @IBAction func confirm() {
+        performSegue(withIdentifier: "waitForConfirmation", sender: self)
     }
     
-    @IBAction func skip(_ sender: Any) {
-        self.view.endEditing(true)
-        self.dismiss(animated: true, completion: nil)
+    @IBAction func textFieldEditingChanged(_ sender: UITextField) {
```
Comment (mathiasquintero) [inline TUM Campus App/LoginViewController.swift]: I love it!

File: TUM Campus App/LoginViewController.swift [hunk 2]
```diff
@@ -13,29 +13,94 @@ class LoginViewController: UIViewController {
...
+            }
+        } else if text.isEmpty {
+            guard let previousTextField = previousTextField else { return }
+
+            previousTextField.becomeFirstResponder()
+        }
+    }
+
+    func textFieldContentsAreValid() -> Bool {
+        guard let firstText = firstTextField.text, firstText.count == 2,
```
Comment (mathiasquintero) [inline TUM Campus App/LoginViewController.swift]: Good catch

File: TUM Campus App/LoginViewController.swift [hunk 3]
```diff
@@ -13,29 +13,94 @@ class LoginViewController: UIViewController {
...
+        textField.layer.borderWidth = isHighlighted ? 1 : 0
+        textField.layer.borderColor = isHighlighted ? UIColor.red.cgColor : UIColor(hexString: "0xC7C7CD").cgColor
+    }
 }
 
 extension LoginViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        firstTextField.delegate = self
```
Comment (mathiasquintero) [inline TUM Campus App/LoginViewController.swift]: Did you set these through the Storyboard, or am I missing something?

**Comment by mathiasquintero** [comment]
Please update the Xcode image to 9.0 in the `.travis.yml`

File: TUM Campus App/LoginViewController.swift (hunk 3 above)
Comment (larryzuo) [reply TUM Campus App/LoginViewController.swift]: They are set through the storyboard. Thats why textFieldEditingChanged is an IBAction that receives textfield events ;)

**Comment by mathiasquintero** [comment]
@mammuth Can you help. You're our Fastlane expert. What did we do last time we had these errors?

**Comment by mammuth** [comment]
We pinned the fastlane version via the Fastfile, but obviously the version that gets installed is different from that.

**Pull request merged**